### PR TITLE
feat: ENT-3804 Add pacing_type and start/end dates to indexed fields in algolia

### PR DESCRIPTION
**Pull request by binodpant**
## Description

Algolia index for enterprise catalog needs some new fields to satisfy requirements of ENT-3518 (pacing_type, start and end dates of the advertised course run)

## Ticket Link

Link to the associated ticket
[ENT-3804](https://openedx.atlassian.net/browse/ENT-3804)

## Post-review

Squash commits into discrete sets of changes

### CODE REVIEW

File: enterprise_catalog/apps/catalog/algolia_utils.py [hunk 1]
```diff
@@ -80,12 +81,12 @@ def _should_index_course(course_metadata):
     """
     course_json_metadata = course_metadata.json_metadata
     advertised_course_run_uuid = course_json_metadata.get('advertised_course_run_uuid')
-    course_runs = course_json_metadata.get('course_runs')
```
Comment (binodpant) [inline enterprise_catalog/apps/catalog/algolia_utils.py:83]: this refactor is optional but I thought would be good to do while I was on it

**Review by adamstankiewicz — COMMENTED** [review]
Code changes look good! Looks like CI is failing, though.

**Comment by binodpant** [comment]
> Code changes look good! Looks like CI is failing, though.

Thanks! yeah finally i? am able to run tests locally so will fix these up

File: enterprise_catalog/apps/catalog/algolia_utils.py [hunk 2]
```diff
@@ -273,6 +275,46 @@ def get_course_card_image_url(course):
...
+    Argument:
+        course (dict)
+
+    Returns:
+        dict: containing key, pacing_type, start and end for the course_run, or None
+    """
+    full_course_run = _get_course_run_by_uuid(course, course.get('advertised_course_run_uuid'))
+    if full_course_run is None:
+        return None
+    course_run = dict(
```
Comment (adamstankiewicz) [inline enterprise_catalog/apps/catalog/algolia_utils.py]: Does creating the dict with `{}` not work here? Curious for the use of `dict()`.

**Review by adamstankiewicz — APPROVED** [review]
LGTM

File: enterprise_catalog/apps/catalog/algolia_utils.py (hunk 2 above)
Comment (binodpant) [reply enterprise_catalog/apps/catalog/algolia_utils.py]: ah no I guess I just lost track of this I had wanted to do that.. made this change!

**Pull request merged**